client/transaction-pool/src/tests/api.rs

```diff
@@ -0,0 +1,178 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Test ChainApi
+
+use crate::*;
+use codec::Encode;
+use parking_lot::RwLock;
+use sp_runtime::{
+	generic::{self, BlockId},
+	traits::{BlakeTwo256, Hash as HashT},
+	transaction_validity::{TransactionValidity, ValidTransaction, TransactionValidityError, InvalidTransaction},
+};
+use std::collections::HashSet;
+use substrate_test_runtime_client::{
+	runtime::{AccountId, Block, BlockNumber, Extrinsic, Hash, Header},
+	AccountKeyring::*,
+};
+
+#[derive(Default)]
+pub struct ChainState {
+	pub block_by_number: HashMap<BlockNumber, Vec<Extrinsic>>,
+	pub block_by_hash: HashMap<Hash, Vec<Extrinsic>>,
+	pub header_by_number: HashMap<BlockNumber, Header>,
+	pub nonces: HashMap<AccountId, u64>,
+	pub invalid_hashes: HashSet<Hash>,
+}
+
+pub struct TestApi {
+	pub valid_modifier: RwLock<Box<dyn Fn(&mut ValidTransaction) + Send + Sync>>,
```

Comment:
Maybe better add a method to alter that and keep all the fields private? Now it's a bit inconsistent.